frontend/js/src/explore/artist-similarity/artist-search/SearchBox.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useMemo, useState } from "react";
+import { faSearch, faMinus, faPlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { throttle } from "lodash";
+import { toast } from "react-toastify";
+import { ToastMsg } from "../../../notifications/Notifications";
+import GlobalAppContext from "../../../utils/GlobalAppContext";
+
+interface SearchBoxProps {
+  currentSimilarArtistsLimit: number;
+  onSimilarArtistsLimitChange: (similarArtistsLimit: number) => void;
+  onArtistChange: (artist_mbid: string) => void;
+}
+
+type ArtistTypeSearchResult = {
+  name: string;
+  id: string;
+  type?: string;
+  country?: string;
+};
+
+function SearchBox({
+  currentSimilarArtistsLimit,
+  onSimilarArtistsLimitChange,
+  onArtistChange,
+}: SearchBoxProps) {
+  const { APIService } = React.useContext(GlobalAppContext);
+  // State to store the search results (list of artists)
+  const [searchResults, setSearchResults] = useState<
+    Array<ArtistTypeSearchResult>
+  >([]);
+  const [searchQuery, setSearchQuery] = useState<string>("");
+  const [openDropdown, setOpenDropdown] = useState<boolean>(false);
+
+  const getArtists = useMemo(() => {
+    return throttle(
+      async (query: string) => {
+        if (query.length && query.trim().length) {
+          try {
+            const results = await APIService.artistLookup(query);
+            const { artists } = results;
+            setSearchResults(artists);
+            setOpenDropdown(true);
+          } catch (error) {
+            setSearchResults([]);
+            toast.error(
+              <ToastMsg
+                title="Search Error"
+                message={typeof error === "object" ? error.message : error}
+              />,
+              { toastId: "error" }
+            );
+          }
+        }
+      },
+      800,
+      { leading: false, trailing: true }
+    );
+  }, []);
+
+  // Lookup the artist based on the query
+  const handleQueryChange = async (query: string) => {
+    setSearchQuery(query);
+    await getArtists(query);
+  };
+
+  // Handle button click on an artist in the dropdown list
+  const handleButtonClick = (artist: ArtistTypeSearchResult) => {
+    onArtistChange(artist.id);
+    setSearchQuery(artist.name);
+  };
+  const increment = () => {
+    onSimilarArtistsLimitChange(currentSimilarArtistsLimit + 1);
+  };
+  const decrement = () => {
+    onSimilarArtistsLimitChange(currentSimilarArtistsLimit - 1);
+  };
+
+  return (
+    <form className="user-inputs-container" autoComplete="off">
+      <div
+        className="artist-input-container"
+        onFocus={() => searchResults.length && setOpenDropdown(true)}
+        onBlur={() => setTimeout(() => setOpenDropdown(false), 200)}
+      >
+        <div className="searchbox-container">
+          <input
```

Comment:
Usually everywhere on the website we use the bootstrap css class `form-control` on all the input elements to homogenize the styling.
That might be at odds with the custom css for these inputs (i.e. where you define a fixed height for example)

Same goes for the other inputs in this page

Reply:
Fixed